### PR TITLE
Add tags to build monitor

### PR DESCRIPTION
**Pull request by adiaaida**
For Validate-DotNet notifications, we want to be able to distinguish builds via how the builds were tagged, and we want to be able to open issues in multiple different repos. This change adds the optional field Tags to BuildDefinition to allow us to specify tags that we would like to match on. It also turns Issues into an array, so that we can have multiple repos that we want to create issues in, and only opens the issues for a build in the linked issue via the Id field.

### CODE REVIEW

**Review by alexperovich — APPROVED** [review]

**Comment by adiaaida** [comment]
Added tests for AzurePipelineMonitor. @ChadNedzlek PTAL

File: src/DotNet.Status.Web.Tests/AzurePipelinesControllerTests.cs [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+        [Test]
+        public async Task BuildCompleteBuildHasNoTags()
+        {
+            var buildEvent = new AzurePipelinesController.AzureDevOpsEvent<AzurePipelinesController.AzureDevOpsMinimalBuildResource>(){
+                Resource = new AzurePipelinesController.AzureDevOpsMinimalBuildResource() {
+                    Id = 123456,
+                    Url = "test-build-url"
+                },
+                ResourceContainers = new AzurePipelinesController.AzureDevOpsResourceContainers(){
+                    Collection = new AzurePipelinesController.HasId() {
```
Comment (ChadNedzlek) [inline src/DotNet.Status.Web.Tests/AzurePipelinesControllerTests.cs:38]:
```suggestion
                    Collection = new AzurePipelinesController.HasId {
```

File: src/DotNet.Status.Web.Tests/AzurePipelinesControllerTests.cs [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+                    Account = new AzurePipelinesController.HasId() {
+                        Id = "test-account-id"
+                    },
+                    Project = new AzurePipelinesController.HasId() {
+                        Id = "test-project-id"
+                    }
+                }
+            };
+
+            var build = new JObject
```
Comment (ChadNedzlek) [inline src/DotNet.Status.Web.Tests/AzurePipelinesControllerTests.cs:50]: We don't use all these values, do we?  We should only send the bits we care about. :-)

Comment (adiaaida) [reply src/DotNet.Status.Web.Tests/AzurePipelinesControllerTests.cs:50]: These are actually all used somewhere in the code path.

File: src/DotNet.Status.Web.Tests/AzurePipelinesControllerTests.cs (hunk 1 above)
Comment (adiaaida) [reply src/DotNet.Status.Web.Tests/AzurePipelinesControllerTests.cs:38]: Fancy

File: src/DotNet.Status.Web.Tests/AzurePipelinesControllerTests.cs [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+using NUnit.Framework;
+using Octokit;
+
+namespace DotNet.Status.Web.Tests
+{
+
+    [TestFixture]
+    public class AzurePipelinesControllerTests
+    {
+        public static string TestTeamsWebHookUri = "https://example.teams/webhook/sha";
```
Comment (ChadNedzlek) [inline src/DotNet.Status.Web.Tests/AzurePipelinesControllerTests.cs:25]: These should probably be const, or at least readonly. Going to have some really weird test behaviors if a test change ones of these, I bet you'll get surprising results. :-)

Comment (adiaaida) [reply src/DotNet.Status.Web.Tests/AzurePipelinesControllerTests.cs:25]: I don't actually think I used that. That was a copy paste from one of Alex's tests. Will remove

**Review by ChadNedzlek — APPROVED** [review]

**Pull request merged**